packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart

```diff
@@ -108,7 +108,17 @@ class FirebaseFirestore extends FirebasePluginPlatform {
   void useFirestoreEmulator(String host, int port, {bool sslEnabled = false}) {
     if (kIsWeb) {
       // use useEmulator() API for web as settings are set immediately unlike native platforms
-      _delegate.useEmulator(host, port);
+      try {
+        _delegate.useEmulator(host, port);
+      } catch (e) {
+        final String message = (e as dynamic).message;
+        // this catches FirebaseError from web that occurs after hot reloading & hot restarting
+        if (!message.contains(
```

Comment:
Could we not hook onto the error code here? Will be the precondition one.

Reply:
That's fine - I'd just rather match against a code rather than a message which could change.